tests/epyccel/modules/augassign.py

```diff
@@ -0,0 +1,155 @@
+# pylint: disable=missing-function-docstring, missing-module-docstring/
+
+from pyccel.decorators import types
+
+# +=
+
+@types('int[:]')
+def augassign_add_1d_int(a):
+    b = a
+    b += 42
+    return b[0]
+
+@types('float[:]')
+def augassign_add_1d_float(a):
+    b = a
+    b += 4.2
+    return b[0]
+
+@types('complex[:]')
+def augassign_add_1d_complex(a):
+    b = a
+    b += (4.0 + 2.0j)
+    return b[0]
+
+@types('int[:,:]')
+def augassign_add_2d_int(a):
+    b = a
+    b += 42
+    return b[0][0]
+
+@types('float[:,:]')
+def augassign_add_2d_float(a):
+    b = a
+    b += 4.2
+    return b[0][0]
+
+@types('complex[:,:]')
+def augassign_add_2d_complex(a):
+    b = a
+    b += (4.0 + 2.0j)
+    return b[0][0]
+
+# -=
+
+@types('int[:]')
+def augassign_sub_1d_int(a):
+    b = a
+    b -= 42
+    return b[0]
+
+@types('float[:]')
+def augassign_sub_1d_float(a):
+    b = a
+    b -= 4.2
+    return b[0]
+
+@types('complex[:]')
+def augassign_sub_1d_complex(a):
+    b = a
+    b -= (4.0 + 2.0j)
+    return b[0]
+
+@types('int[:,:]')
+def augassign_sub_2d_int(a):
+    b = a
+    b -= 42
+    return b[0][0]
+
+@types('float[:,:]')
+def augassign_sub_2d_float(a):
+    b = a
+    b -= 4.2
+    return b[0][0]
+
+@types('complex[:,:]')
+def augassign_sub_2d_complex(a):
+    b = a
+    b -= (4.0 + 2.0j)
+    return b[0][0]
+
+# *=
+
+@types('int[:]')
+def augassign_mul_1d_int(a):
+    b = a
+    b *= 42
+    return b[0]
+
+@types('float[:]')
+def augassign_mul_1d_float(a):
+    b = a
+    b *= 4.2
+    return b[0]
+
+@types('complex[:]')
+def augassign_mul_1d_complex(a):
+    b = a
+    b *= (4.0 + 2.0j)
+    return b[0]
```

Comment:
Checking the first elements is enough?

Reply:
nvm, you are using `array_equal`, but that means the return is not necessary?

Reply:
It's use now in the test functions: https://github.com/pyccel/pyccel/pull/1286/commits/7f2400caf3017a9be96cca9df9f72595526e50ac